docs/source/guide/build_pipelines/index.rst

```diff
@@ -1,8 +1,207 @@
+      lineapy.save(df, "iris_preprocessed")
+
+
+   def iris_model():
+      art_df_processed = lineapy.get("iris_preprocessed")
+      df_processed = art_df_processed.get_value()
+      mod = LinearRegression()
+      mod.fit(
+         X=df_processed[["petal.width", "d_versicolor", "d_virginica"]],
+         y=df_processed["sepal.width"],
+      )
+      lineapy.save(mod, "iris_model")
+
+We can see that LineaPy used artifacts to automatically 1) slice their code to retain only essential operations and 2) package the sliced code into importable functions.
+
+And we see ``demo_airflow_pipeline_dag.py`` automatically composing an Airflow DAG with these functions:
+
+.. code:: python
+
+   import os
+
+   import demo_airflow_pipeline
+   from airflow import DAG
+   from airflow.operators.python_operator import PythonOperator
+   from airflow.utils.dates import days_ago
+
+   default_dag_args = {"owner": "airflow", "retries": 2, "start_date": days_ago(1)}
+
+   dag = DAG(
+      dag_id="demo_airflow_pipeline_dag",
+      schedule_interval="*/15 * * * *",
+      max_active_runs=1,
+      catchup=False,
+      default_args=default_dag_args,
+   )
+
+
+   iris_preprocessed = PythonOperator(
+      dag=dag,
+      task_id="iris_preprocessed_task",
+      python_callable=demo_airflow_pipeline.iris_preprocessed,
+   )
+
+   iris_model = PythonOperator(
+      dag=dag,
+      task_id="iris_model_task",
+      python_callable=demo_airflow_pipeline.iris_model,
+   )
+
+
+   iris_preprocessed >> iris_model
+
+Next, we see ``demo_airflow_pipeline_requirements.txt`` listing dependencies for running the pipeline:
+
+.. code:: none
+
+   matplotlib==3.5.1
+   numpy==1.21.5
+   lineapy==0.0.5
+
+Finally, we have the automatically generated Dockerfile (``demo_airflow_pipeline_Dockerfile``), which facilitates pipeline execution:
+
+.. code:: docker
+
+   FROM apache/airflow:latest
+
+   RUN mkdir /tmp/installers
+   WORKDIR /tmp/installers
+
+   # copy all the requirements to run the current dag
+   COPY ./demo_airflow_pipeline_requirements.txt ./
+   # install the required libs
+   RUN pip install -r ./demo_airflow_pipeline_requirements.txt
+
+   WORKDIR /opt/airflow/dags
+   COPY . .
+
+   WORKDIR /opt/airflow
+
+   CMD [ "standalone" ]
+
+Testing Locally
+---------------
+
+With these automatically generated files, we can quickly test running the pipeline locally.
+First, we run the following command to build a Docker image:
+
+.. code:: bash
+
+    docker build -t <image_name> . -f <pipeline_name>_Dockerfile
+
+where ``<pipeline_name>_Dockerfile`` is the name of the automatically generated Dockerfile and
+``<image_name>`` is the image name of our choice.
+
+We then stand up a container instance with the following command:
+
+.. code:: bash
+
+    docker run -it -p 8080:8080 <image_name>
+
+In the current example where we set ``framework="AIRFLOW"``, this will result in an Airflow instance
+with an executable DAG in it.
```

Comment:
@lionsardesai I refactored "Running Airflow DAGs" section into this subsection. Please let me know if you have questions about this.